modules/apps/frontend-taglib/frontend-taglib-chart/src/main/resources/META-INF/resources/resources.js

```diff
@@ -0,0 +1,11 @@
+import mainCssSource from "clay-charts/lib/css/main.css";
+import tilesSvgSource from "clay-charts/lib/svg/tiles.svg";
+
+const node = document.createElement('code');
+
+node.innerHTML = `
```

Comment:
This is where we write the resources to the HTML (client side).

We cannot do it easily from server side because webpack doesn't emit the assets. Also, I cannot see any good way to do this transparently as we did in bundler 2, though we may provide some infra helper to move this from the app side to infra.

Reply:
This feels like a different issue. To me, the problems here are:
- How do we reference a bundled static resource from some Java code.
- How do we bundle a specific resource from an NPM module?

In the past, we bundled (almost) everything by default and made some modifications to the paths which meant we had to provide helpers to find the right path.

Now we don't bundle anything that is not in the path of an entry point for webpack, but neither does gradle for anything outside of the default module layout. We've come to rely on `-includeresource` or similar in `bnd.bnd` so we if we can use that to solve the bundling issue then we would only need to worry about the paths, but if those resources are not part of any modification now, that should also be straightforward.

This is a complete different problem from loaders, which we should also address and configure properly, so code that's already requesting assets from JS can continue to do so under our platform.

Reply:
Using `-includeresource` would be great for this case. Sadly, because all the JS deps are in `liferay-portal/modules/node_modules` we cannot reference them in a clean way.

However, we may try to tweak webpack to generate fixed file names in the output folder and that would suffice. We would still need to do the `import` in the JS code, but we wouldn't need to add the nodes to the HTML, which is the ugliest part of the solution IMO.

Reply:
Let me check if it is easy to give resources a predictable name and I can send another commit to use `-includeresource` so that we can close this. If it is not easy I will close the PR and defer it.

Reply:
Here it is -> https://webpack.js.org/guides/asset-modules/#custom-output-filename
And -> https://webpack.js.org/configuration/output/#outputfilename

@jbalsas What if we generate predictable names for both `main.css` and `tiles.svg` and include them with `includeresource` to unblock this?

Reply:
Let's just table this specific use case for later and focus on the rest. Since this only affects 1 module right now, I'd rather wait and push forward with the replacement where we can before adding more complexity to the build.